include/librpmi.h

```diff
@@ -783,6 +796,230 @@ struct rpmi_service_group {
+ */
+enum rpmi_error rpmi_clock_get_attrs(struct rpmi_clock_module *clock_mod,
+				     rpmi_uint32_t clock_id,
+				     struct rpmi_clock_attrs *attrs);
+
+/**
+ * @brief Set clock state
+ *
+ * @param[in] clock_mod		pointer to clock_module instance
+ * @param[in] clock_id		clock id
+ * @param[in] state		new clock state
+ * @return enum rpmi_error
+ */
+enum rpmi_error rpmi_clock_set_state(struct rpmi_clock_module *clock_mod,
+				     rpmi_uint32_t clock_id,
+				     enum rpmi_clock_state state);
+
+/**
+ * @brief Get clock current state
+ *
+ * @param[in] clock_mod		pointer to clock_module instance
+ * @param[in] clock_id		clock id
+ * @param[in] state		pointer to state buffer
+ * @return enum rpmi_error
+ */
+enum rpmi_error rpmi_clock_get_state(struct rpmi_clock_module *clock_mod,
+				     rpmi_uint32_t clock_id,
+				     enum rpmi_clock_state *state);
+
+/**
+ * @brief Set clock rate
+ *
+ * @param[in] clock_mod		pointer to clock_module instance
+ * @param[in] clock_id		clock id
+ * @param[in] match		clock rate match mode
+ * @param[in] rate		new clock rate in Hertz
+ * @return enum rpmi_error
+ */
+enum rpmi_error rpmi_clock_set_rate(struct rpmi_clock_module *clock_mod,
+				    rpmi_uint32_t clock_id,
+				    enum rpmi_clock_rate_match match,
+				    rpmi_uint64_t rate);
+
+/**
+ * @brief Get clock rate
+ *
+ * @param[in] clock_mod		pointer to clock_module instance
+ * @param[in] clock_id		clock id
+ * @param[in] rate		pointer to rate buffer
+ * @return enum rpmi_error
+ */
+enum rpmi_error rpmi_clock_get_rate(struct rpmi_clock_module *clock_mod,
+				    rpmi_uint32_t clock_id,
+				    rpmi_uint64_t *rate);
+
+/**
+ * @brief Create a clock service group instance
+ *
+ * @param[in] clock_mod		pointer to clock module
+ * @return rpmi_service_group *	pointer to RPMI service group instance upon
+ * success and NULL upon failure
+ */
+struct rpmi_service_group *
+rpmi_service_group_clock_create(struct rpmi_clock_module *clock_mod);
+
+/**
+ * @brief Destroy(free) a clock service group instance
+ *
+ * @param[in] group	pointer to RPMI service group instance
+ */
+void rpmi_service_group_clock_destroy(struct rpmi_service_group *group);
+
+/**
+ * @brief Create a clock module
+ *
+ * @param[in] clock_count	Number of clocks
+ * @param[in] clock_tree_data	pointer to list of clocks data
+ * @param[in] ops		pointer to platform specific system reset operations
+ * @param[in] ops_priv		pointer to private data of platform operations
+ * @return rpmi_clock_module *	pointer to RPMI service group instance upon
+ * success and NULL upon failure
+ */
+struct rpmi_clock_module *rpmi_clock_module_create(rpmi_uint32_t clock_count,
+					const struct rpmi_clock_data *clock_tree_data,
+					const struct rpmi_clock_platform_ops *ops,
+					void *ops_priv);
+
+/**
+ * @brief Destroy(free) a clock module instance
+ *
+ * @param[in] clock_mod		pointer to clock module instance
+ */
+void rpmi_clock_module_destroy(struct rpmi_clock_module *clock_mod);
```

Comment:
We can merge rpmi_clock_module_create() and rpmi_clock_module_destroy() into rpmi_service_group_clock_create() and rpmi_service_group_clock_destroy().